Makefile

```diff
@@ -171,6 +179,53 @@ $(BUILT)/$(ADD-ON)/chrome.manifest: $(ADD-ON)/jar.mn
 	sed -e "s/^% //g" | sed -e "s/%/chrome\//g" \
 	> $(BUILT)/$(ADD-ON)/chrome.manifest
 
+# In this RSYNC, the order of exclude and includes generally matters.
+# The items below are:
+# - Don't copy prefs.js as we generate that in the dependencies.
+# - Copy all the relevant js, html, css etc files that we need.
+# - Copy the add-on "support files".
+# - Exclude all other files.
+# - Exclude all test files.
+# - Exclude and editor temp files.
+RSYNC_DIST := $(RSYNC) -r \
+  --exclude="**/prefs.js" \
+  --include="*.jsm" --include="*.js" \
+	--include="*.html" --include="*.css"  \
+	--include="*.ogg" --include="*.svg" --include="*.png" --include="*.gif" \
+	--include="*.properties" --include="*.manifest" --include="*.rdf" \
+	--exclude="*.*" --exclude "**/test" --exclude "*~"
+
+.PHONY: add-on_dist
+add-on_dist: $(DIST)/add-on/chrome/content/preferences/prefs.js
+	mkdir -p $(DIST)/add-on
+	# Copy just the files we need.
+	$(RSYNC_DIST) $(BUILT_ADD_ON)/ $(DIST)/add-on/
+	mv $(DIST)/add-on/chrome/content/shared/vendor/react-prod.js \
+	   $(DIST)/add-on/chrome/content/shared/vendor/react.js
+
+.PHONY: standalone_dist
+standalone_dist:
+	mkdir -p dist
+	$(RSYNC) content/* dist
+	NODE_ENV="production" $(NODE_LOCAL_BIN)/webpack \
+		-p -v --display-errors
+	sed 's#webappEntryPoint.js#js/standalone.js#' \
+		< content/index.html > dist/index.html
+
+XPI_FILE := built/$(XPI_NAME)
+
+# so we can type "make xpi" without depend on the file directly
+.PHONY: xpi
+xpi: $(XPI_FILE)
+
+$(XPI_FILE): $(REPO_BIN_DIR)/build_extension.sh build
+	@$(REPO_BIN_DIR)/build_extension.sh $(@F) built add-on
```

Comment:
Not sure if it's worthwhile to skip the `$(@F)` and just use `$(XPI_NAME)` to match the line below for dist_xpi with the only differences between the lines being `$(BUILT)` and `$(DIST)`